src/components/activity/ActivityController.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+
+#include "components/datetime/DateTimeController.h"
+#include "components/settings/Settings.h"
+#include "components/motion/MotionController.h"
+
+namespace Pinetime {
+  namespace Controllers {
+    class ActivityController {
+    public:
+      ActivityController(Controllers::DateTime& dateTimeController,
+                         Controllers::Settings& settingsController,
+                         Controllers::MotionController& motionController);
+
+      void UpdateSteps();
+      bool ShouldNotify();
+
+    private:
+      Controllers::DateTime& dateTimeController;
+      Controllers::Settings& settingsController;
+      Controllers::MotionController& motionController;
```

Comment:
I agree with you for both the Settings controller and the Motion controller, but to me it seems that having the number of minutes since the hour in UTC is a bit of a strange parameter to a function. Do you think that it makes sense to do that? It would also mean moving the local time to UTC calculation to the caller, which seems a bit odd to me.

Reply:
I don't see how local time to UTC calculation belongs in this class either. If it needs UTC, it should be passed UTC. That's all it needs right? There should just be a function to get UTC time from DateTimeController.

Reply:
Actually, if you use UTC, depending on the timezone, the notification may come at 0 or 30 minutes in local time, or 15 or 45 in another timezone right? I still don't fully understand the code, but that seems a bit weird. Maybe it should just use local time.

Reply:
If someone is in a timezone with a 15 or 45 minute offset from UTC, they will receive the notification at 15 and 45 past the hour, yes. I agree that's weird, and I think the issues from using local time are infrequent enough that I'll switch to local time. The issues with local time are that if you switch to a fractional timezone, the index will be offset and so the buffer accesses will be strange. This would also happen if you change the time by a non-hour amount.